cliquet_docs/reference/permission.rst

```diff
@@ -105,23 +183,162 @@ To do so, subclass the default ``AuthorizationPolicy`` and add a specific
     from zope.interface import implementer
 
     @implementer(IAuthorizationPolicy)
-    class AuthorizationPolicy(authorization.AuthorizationPolicy):
-        def get_bound_permissions(self, *args, **kwargs):
-        """Callable that takes an object ID and a permission and returns
-        a list of tuples (<object id>, <permission>)."""
-            return build_permissions_set(*args, **kwargs)
+    class MyAuthz(authorization.AuthorizationPolicy):
+        def get_bound_permissions(self, permission_object_id, permission):
+            related = [(permission_object_id, permission)]
+            if permission_object_id.startswith('/articles'):
+                related.append((permission_object_id + '/categories', permission))
+            return related
+
+This would require forcing the setting ``multiauth.authorization_policy = myapp.authz.MyAuthz``.
+
+
+Manipulate permissions
+----------------------
+
+One way of achieving dynamic permissions is to manipulate the permission backend
+manually.
+
+For example, in some admin view:
+
+.. code-block:: python
+
+    def admin_view(request):
+        permission = request.registry.permission
+
+        some_user_id = request.POST['user_id']
+        perm_object_id = '/{collection}'
+        permission.add_principal_to_ace(perm_object_id,
```

Comment:
I commented on outdated diff so I post it here again so you won't miss it:

It is very relevant, I was just thinking it will be better with a comment explaining what does this view. As a beginner I like to see code explained and not try to guess what the example does. It is possible that the reader is not familiar with acl or the permission mechanism.
I was suggesting you could add  `#  user_id can now create record on {collection}`. I like repetition via comment but that's my opinion